builder.go

```diff
@@ -175,25 +203,57 @@ func containsEntry(metRequires []string, entry BuildPlanEntry) bool {
 	return false
 }
 
-type processMap map[string]launch.Process
+type processMap struct {
+	typeToProcess map[string]launch.Process
+	defaultType   string
+}
+
+func newProcessMap() processMap {
+	return processMap{
+		typeToProcess: make(map[string]launch.Process),
+		defaultType:   "",
+	}
+}
 
-func (m processMap) add(l []launch.Process) {
-	for _, proc := range l {
-		m[proc.Type] = proc
+// This function adds the processes from listToAdd to processMap
+// it sets m.defaultType to the last default process
+// if a non-default process overrides a default process, it returns its type and unset m.defaultType
+func (m *processMap) add(listToAdd []launch.Process) string {
+	result := ""
+	for _, procToAdd := range listToAdd {
+		if procToAdd.Default {
+			m.defaultType = procToAdd.Type
+			result = ""
+		} else {
+			existingProc, ok := m.typeToProcess[procToAdd.Type]
+			if ok && existingProc.Type == m.defaultType { // existingProc.Default = true
```

Comment:
```suggestion
			procToAdd.Type == m.defaultType { // non-default process overrides a default process
```
I **think** this will achieve the same result

Reply:
This function might read a bit easier if you returned the entire warning instead of just the type that was replaced.

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/520/commits/7b24df17bb72f57cfba08d780795cbaba5aaddca.